control-plane/api-gateway/common/translation.go

```diff
@@ -299,10 +303,36 @@ func (t ResourceTranslator) translateHTTPFilters(filters []gwv1beta1.HTTPRouteFi
 			filter.URLRewrite.Path.Type == gwv1beta1.PrefixMatchHTTPPathModifier {
 			urlRewrite = &api.URLRewrite{Path: DerefStringOr(filter.URLRewrite.Path.ReplacePrefixMatch, "")}
 		}
+
+		if filter.ExtensionRef != nil {
+			//get crd from resources map
+			crdFilter, exists := resourceMap.GetExternalFilter(*filter.ExtensionRef)
+			if !exists {
+				//doesn't exist yet, hope this gets handled next reconcillation loop
```

Comment:
can you update this comment assuming the statement below is actually true (I'm still fine with the defensive programming though)? something like?
```suggestion
				// this should never be the case because we only translate a route if it's actually valid, and if we're missing filters during the validation step, then we won't get here
```